include/swift/Sema/ConstraintSystem.h

```diff
@@ -3404,6 +3420,12 @@ class ConstraintSystem {
   void recordCallAsFunction(UnresolvedDotExpr *root, ArgumentList *arguments,
                             ConstraintLocator *locator);
 
+  /// Record root, value, and declContext of keypath expression for use across
+  /// constraint system.
+  void recordKeyPath(KeyPathExpr *keypath,
+                     std::tuple<TypeVariableType *, TypeVariableType *,
+                                DeclContext *> const &info);
```

Comment:
Let's make this 3 separate parameters instead, we can construct a tuple internally.